### ISSUE TITLE
Improving YOLOv4 over YOLOv3

### ISSUE THREAD

**Issue by SalahAdDin** [AlexeyAB/darknet#5619]
Hi, I just taking advantage of the new release; it says it makes training suitable for GPUs.

Well, I trained YOLOv3 in Colab, and for this dataset(current training YOLOv4), it took around 9 hours... so, is it normal that training YOLOv4 would take around 20 hours?

**Comment by keita-honsho** [comment]
Hi.

I'm learning v4 on RTX2080, currently iteration 179815 / 500500 and approx. time left is 824hours. I'm in trouble.

**Comment by WongKinYiu** [comment]
`width=608` and `height=608` by default in `yolo4_custom` configuration file is for testing.

The default setting for training `yolov4` is `width=512` and `height=512`.
If you feel your training time is too long, you can set `width`, `heights`, and `anchors` same as `yolov3`.

**Comment by SalahAdDin** [comment]
> `width=608` and `height=608` by default in `yolo4_custom` configuration file is for testing.
> 
> The default setting for training `yolov4` is `width=512` and `height=512`.
> If you feel your training time is too long, you can set `width`, `heights`, and `anchors` same as `yolov3`.

Yeah, but, in the paper, its says the main goal is making training suitable for single common GPU, and, since this values i think it is not fulfill this goal.

**Comment by AlexeyAB** [comment]
> so, is it normal that training YOLOv4 would take around 20 hours?

Yes, if you want high accuracy, then there is nothing better than YOLOv4.

> Around 31 hours for training?

Yes, it takes a little time compared to other modern neural networks, where you will train for several months, and will get much lower accuracy. So yolov4 training is several times faster than EfficientDet, ASFF-800, ...

Also EfficientDet requires 512 GB-VARM for training with mini-batch-size=128, otherwise you will get lower accuracy than claimed:

EfficientDet: https://arxiv.org/abs/1911.09070 

> Synchronized batch normalization is added after every convolution with batch norm decay 0.99 and epsilon 1e-3. 
> Each model is trained 300 epochs with **batch total size 128** on 32 TPUv3 cores.

https://cloud.google.com/tpu/docs/types-zones#europe

> v3-32 TPU type (v3) – 32 TPU v3 cores – **512 GiB Total TPU memory**
